dialogue-apache-hc5-client/src/test/java/com/palantir/dialogue/hc5/NoResponseTest.java

```diff
@@ -85,6 +113,82 @@ public void testConnectionClosedAfterDelay() {
         }
     }
 
+    @Test
+    public void testLargePayload() {
+        String randomHostname = UUID.randomUUID().toString();
+
+        // Set up Undertow server with a large payload
+        String largePayload = "a".repeat(MAX_BYTES * 2); // This payload should exceed the maximum allowed size
+        Undertow server = createUndertowServerWithPayload(largePayload);
+        server.start();
+
+        try {
+            ReloadingFactory factory = DialogueClients.create(Refreshable.only(ServicesConfigBlock.builder()
+                            .defaultSecurity(TestConfigurations.SSL_CONFIG)
+                            .putServices(
+                                    "foo",
+                                    PartialServiceConfiguration.builder()
+                                            .addUris(getUri(server, randomHostname))
+                                            .build())
+                            .build()))
+                    .withUserAgent(TestConfigurations.AGENT)
+                    .withDnsResolver(hostname -> {
+                        if (randomHostname.equals(hostname)) {
+                            try {
+                                return ImmutableSet.of(
+                                        InetAddress.getByAddress(randomHostname, new byte[] {127, 0, 0, 1}));
+                            } catch (UnknownHostException ignored) {
+                                // fall-through
+                            }
+                        }
+                        return ImmutableSet.of();
+                    });
+
+            ReloadingFactory reloadingFactory = factory.withRuntime(new ConjureRuntime() {
+                @Override
+                public BodySerDe bodySerDe() {
+                    return new ConjureBodySerDe(
```

Comment:
Very much a WIP. I've made many classes that ought not to be public, public, in an attempt to get my LimitSizeEncoding serializer to get called. Going to step through the code to figure out why the new encoder is not getting used.